src/win/util.c

```diff
@@ -1664,26 +1664,43 @@ int uv_os_unsetenv(const char* name) {
 
 
 int uv_os_gethostname(char* buffer, size_t* size) {
-  char buf[UV_MAXHOSTNAMESIZE];
+  WCHAR* buf;
   size_t len;
+  char* utf8_str;
+  int convert_result;
 
   if (buffer == NULL || size == NULL || *size == 0)
     return UV_EINVAL;
 
   uv__once_init(); /* Initialize winsock */
 
-  if (gethostname(buf, sizeof(buf)) != 0)
+  buf = uv__malloc(sizeof(WCHAR) * UV_MAXHOSTNAMESIZE);
```

Comment:
Why aren't we declaring `buf` on the stack like this?
```c
WCHAR buf[UV_MAXHOSTNAMESIZE];
```

Reply:
Done